protocol/contracts/nexus/Relic.sol

```diff
@@ -0,0 +1,344 @@
+    }
+
+    //------- Public -------//
+
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        override(IERC165, ERC721, ERC721Enumerable)
+        returns (bool)
+    {
+        return super.supportsInterface(interfaceId);
+    }
+
+    function onERC1155Received(
+        address,
+        address,
+        uint256,
+        uint256,
+        bytes memory
+    ) public virtual override returns (bytes4) {
+        return this.onERC1155Received.selector;
+    }
+
+    function onERC1155BatchReceived(
+        address,
+        address,
+        uint256[] memory,
+        uint256[] memory,
+        bytes memory
+    ) public virtual override returns (bytes4) {
+        return this.onERC1155BatchReceived.selector;
+    }
+
+    function onERC721Received(
+        address,
+        address,
+        uint256,
+        bytes memory
+    ) public virtual returns (bytes4) {
+        return this.onERC721Received.selector;
+    }
+
+    function getRelicXP(uint256 _relicId) public view returns (uint256) {
+        require(_exists(_relicId), "This Relic doesn't exist");
+        return relicXP[_relicId];
+    }
+
+    function tokenURI(uint256 _relicId)
+        public
+        view
+        virtual
+        override(ERC721, ERC721URIStorage)
+        returns (string memory)
+    {
+        require(_exists(_relicId));
+        return
+            string(
+                abi.encodePacked(
+                    BASE_URIS[_getRarity(_relicId)],
+                    uint256(enclaves[_relicId]).toString()
+                )
+            );
+    }
+    // get Shard balance for a specific shard Id from a Relic
+    function getBalance(uint256 _relicId, uint256 _tokenId)
+        external
+        view
+        returns (uint256)
+    {
+        require(_exists(_relicId), "This Relic doesn't exist.");
+        return balances[_relicId][_tokenId];
+    }
+
+    function getBalanceBatch(uint256 _relicId, uint256[] memory _tokenId)
+        external
+        view
+        returns (uint256[] memory)
+    {
+        require(_exists(_relicId), "This Relic doesn't exist.");
+
+        uint256[] memory balancesToReturn = new uint256[](_tokenId.length);
+        for (uint256 i = 0; i < _tokenId.length; i++) {
+            balancesToReturn[i] = balances[_relicId][_tokenId[i]];
+        }
+
+        return balancesToReturn;
+    }
+
+    // fetch Rarity and Enclave from a single Relic
+    function getRelicInfos(uint256 _relicId)
+        external
+        view
+        returns (Rarity, Enclave)
+    {
+        return (_getRarity(_relicId), enclaves[_relicId]);
+    }
+
+    //------- Internal -------//
+
+    function _beforeTokenTransfer(
+        address from,
+        address to,
+        uint256 firstTokenId,
+        uint256 batchSize
+    ) internal override(ERC721, ERC721Enumerable) whenNotPaused {
+        super._beforeTokenTransfer(from, to, firstTokenId, batchSize);
+    }
+
+    function _burn(uint256 tokenId)
```

Comment:
Nothing calls burn as far as i can tell - is there meant to be a mechanism where nexus/users can burn tokens?

Reply:
Ah i missed that this implements `ERC721Burnable`. So ignore this.

Any bad consequences if a user does burn their Relic? For example shards/balances/etc would be lost too -- would this mess up balances?